src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -2896,6 +2899,15 @@ module EstablishTypeDefinitionCores =
 
                 // Run InferTyconKind to raise errors on inconsistent attribute sets
                 InferTyconKind g (SynTypeDefnKind.Record, attrs, [], [], inSig, true, m) |> ignore
+                
+                if g.langVersion.SupportsFeature(LanguageFeature.EnforceAttributeTargets) then
+                    if hasStructAttr then
+                        if hasRQAAttr then
```

Comment:
Can you please add a comment on why struct and rqa have to be special-cased?

Reply:
Sure. Ideally, this should not be a special case. But the fact that previously were not enforcing  accurate `AttributeTargets` kind of worked for most the targets. So this checks to ensure that we can still use the `RequireQualifiedAccess` on structs.

Reply:
Done.